delfin/drivers/dell_emc/vnx/vnx_block/component_handler.py

```diff
@@ -565,3 +572,340 @@ def list_storage_hosts(self, storage_id):
                 }
                 host_list.append(host_model)
         return host_list
+
+    def collect_perf_metrics(self, storage_id, resource_metrics,
+                             start_time, end_time):
+        metrics = []
+        archive_file_list = []
+        try:
+            if (end_time - start_time) < consts.EXEC_TIME_INTERVAL:
+                LOG.warn("not exe collert, end_time-start_time={}".format(
+                    (end_time - start_time)))
+                return metrics
+            archive_file_list = self._get__archive_file(start_time, end_time)
+            LOG.info("Get archive files: {}".format(archive_file_list))
+            if not archive_file_list:
+                LOG.warn("The required performance file was not found!")
+                return metrics
+            resources_map, resources_type_map = self._get_resources_map(
+                resource_metrics)
+            if not resources_map or not resources_type_map:
+                LOG.warn("Resource object not found!")
+                return metrics
+            performance_lines_map = self._filter_performance_data(
+                archive_file_list, resources_map, start_time, end_time)
+            if not performance_lines_map:
+                LOG.warn("The required performance data was not found!")
+                return metrics
+            metrics = self.create_metrics(storage_id, resource_metrics,
+                                          resources_map, resources_type_map,
+                                          performance_lines_map)
```

Comment:
Please add log info to record storage_id, start_time, end_time, and the length of `metrics` when collect successfully.

Reply:
Modified